pkg/services/publicdashboards/commands/generate_datasources/main_test.go

```diff
@@ -0,0 +1,35 @@
+package generate_datasources
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCanGetCompatibleDatasources(t *testing.T) {
```

Comment:
testing. epic.

Reply:
This test took me a while. So many different answers out there for how to mock and stub things. If you want to mock an http client, then using the `httptest` package is the way to go, allegedly.